File: 2048-build-array-from-permutation/2048-build-array-from-permutation.cs
public class Solution {
    public int[] BuildArray(int[] nums) {
        var n = nums.Length;
        var ans = new int[n];
        for (int i = 0; i < n; ++i)
            ans[i] = nums[nums[i]];
        
        return ans;
    }
}

File: 2048-build-array-from-permutation/2048-build-array-from-permutation.java
class Solution {
    public int[] buildArray(int[] nums) {
        int n = nums.length;
        int[] sms = new int[n];

        for (int i = 0; i < n; ++i)
            sms[i] = nums[nums[i]];
            
        return sms;
    }
}